### ISSUE TITLE
Store `recordsPath` paths relative to `options.context`

### ISSUE THREAD

**Issue by sirlancelot** [webpack/webpack#295]
How difficult would it be to output relative paths for the `recordsPath` file? I feel like it would be useful in scenarios where you want multiple machines to predictably compile modules in the same way every time. This could apply to a dev/prod environment or even a multi-developer environment.

**Comment by jtangelder** [comment]
What's the status of this?

**Comment by sokra** [comment]
Not done yet...

You can do a PR...

**Comment by MPV** [comment]
Hi @sokra, 
We just switched from `OccurenceOrderPlugin` to `recordsPath` (since OccurrenceOrderPlugin gave different output on different machines). 

However, the paths in our records file are still absolute (even after your fix above). 

Hence they still differ on different machines. 

We are using Webpack `1.12.2`.

**Comment by MPV** [comment]
We have now tried switching between `jsx-loader` and `babel-loader` and both gives absolute urls like this:

jsx-loader:
```
"node_modules/jsx-loader/index.js!/Users/vs/Projects/myproject/frontend/react/mixins/translations/index.js": 278,
```
babel-loader:
```
"node_modules/babel-loader/index.js!/Users/carlb/projects/myproject/frontend/react/apps/ntf/entry.jsx": 0,
```

Now suspecting that the needed change might be in webpack anyway. I've looked through your fix from 44cfedd but I haven't yet understood what change to webpack that would be necessary to get relative paths for loaders too.

**Comment by carlbarrdahl** [comment]
@sokra @MPV 
It seems 44cfedd doesn't handle relative paths for loaders. The identifier should split on '!'.

https://github.com/webpack/webpack/blob/44cfedd24e0a96f68dcdcb1c7d25ef4bc6bcc085/lib/RecordIdsPlugin.js#L13-L15

**Comment by sokra** [comment]
yep, I know, I already fixed it but forgot to commit it... Comming soon.

**Comment by MPV** [comment]
@sokra Amazing, thanks!

I noticed the change was only added to `v2.0.0-beta` and we're running `1.x`.

Do you think it might be possible to create a new tag/release with this fix (for example `1.12.3`)?
